modules/objdetect/src/cascadedetect.cpp

```diff
@@ -587,7 +587,7 @@ bool HaarEvaluator::read(const FileNode& node, Size _origWinSize)
     localSize = lbufSize = Size(0, 0);
     if (ocl::haveOpenCL())
     {
-        if (ocl::Device::getDefault().isAMD())
+        if (ocl::Device::getDefault().isAMD() || ocl::Device::getDefault().isIntel())
```

Comment:
Last time I checked it on my MBP15 with Iris Pro, CPU branch was faster than the GPU one. Do you have some statistics regarding OpenCL branch on various Intel's chips?